### PR TITLE
dlab_api - add task processing

### PR DESCRIPTION
**Pull request by Dkhanas**

### CODE REVIEW

File: api/processes_manager.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import subprocess
+import sys
+from threading import Thread
+
+import persistqueue
+
+from dlab_core.infrastructure.repositories import SQLiteRepository, STARTED, DONE
+
+DB_PATH = 'C:\Users\Dmytro_Khanas\projects\dlab'
```
Comment (mediapills) [inline api/processes_manager.py]: package_data in setup.py

File: api/processes_manager.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+num_worker_threads = 2
+
+
+class BaseManager(object):
+    def __init__(self, location=None):
+        if location is None:
+            location = DB_PATH
+        self.location = location
+        self.queue = persistqueue.FIFOSQLiteQueue(location, multithreading=True, auto_commit=False)
+        self.repo = SQLiteRepository(location+'\data.db')
```
Comment (mediapills) [inline api/processes_manager.py]: directory separator from os lib

File: api/processes_manager.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+DB_PATH = 'C:\Users\Dmytro_Khanas\projects\dlab'
+num_worker_threads = 2
+
+
+class BaseManager(object):
+    def __init__(self, location=None):
+        if location is None:
+            location = DB_PATH
+        self.location = location
+        self.queue = persistqueue.FIFOSQLiteQueue(location, multithreading=True, auto_commit=False)
```
Comment (mediapills) [inline api/processes_manager.py]: add TODO move into repo

File: api/processes_manager.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        for i in range(num_worker_threads):
+            t = Thread(target=self.run)
+            t.start()
+
+    def run(self):
+        while True:
+            record_id = self.start_task()
+            try:
+                # TODO: run cli task
+                cmd = 'C:\Users\Dmytro_Khanas\projects\dlab\\test.py "{}"'.format(record_id)
```
Comment (mediapills) [inline api/processes_manager.py]: redesign path

File: api/processes_manager.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+                out, err = p.communicate()
+                if err:
+                    raise Exception(err)
+                print(out, err)
```
Comment (mediapills) [inline api/processes_manager.py]: remove prints

File: api/processes_manager.py (hunk 1 above)
Comment (Dkhanas) [reply api/processes_manager.py]: added

File: api/processes_manager.py (hunk 2 above)
Comment (Dkhanas) [reply api/processes_manager.py]: updated

File: api/processes_manager.py (hunk 3 above)
Comment (Dkhanas) [reply api/processes_manager.py]: moved into repo

File: api/processes_manager.py (hunk 4 above)
Comment (Dkhanas) [reply api/processes_manager.py]: fixed

File: api/processes_manager.py (hunk 5 above)
Comment (Dkhanas) [reply api/processes_manager.py]: removed

**Pull request closed without merging**